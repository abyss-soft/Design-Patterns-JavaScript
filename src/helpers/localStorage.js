export const loadState = () => {
  try {
    const serializedState = JSON.parse(localStorage.getItem('state'));

    return serializedState === null ? undefined : serializedState;
  } catch (e) {
    return undefined;
  }
};

export const saveState = (state) => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (e) {
    return undefined;
  }
};
