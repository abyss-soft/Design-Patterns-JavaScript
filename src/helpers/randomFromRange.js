// [0, max] range
export const randomFromRange = max => Math.floor(Math.random() * max);
