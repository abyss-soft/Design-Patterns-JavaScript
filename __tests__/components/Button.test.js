import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Button from '../../src/components/Button';
import { renderWithLightTheme } from '../helpers';
import { renderWithDarkTheme } from '../helpers';

describe('<Button /> component', () => {
  const mockClick = jest.fn();
  const buttonEl = <Button label="Test Button" id="test" onClick={mockClick} />;
  let button;

  beforeEach(() => {
    button = shallow(buttonEl);
  });

  it('renders without crashing', () => {
    expect(button).toBeTruthy();
  });

  it('has styled-component rendered classes', () => {
    const tree = renderer.create(buttonEl).toJSON();
    expect(tree.props.className).toBeDefined();
  });

  it('renders with a LIGHT theme', () => {
    const tree = renderWithLightTheme(buttonEl).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with a DARK theme', () => {
    const tree = renderWithDarkTheme(buttonEl).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has the passed label', () => {
    const buttonChilren = button.getElement().props.children;

    expect(buttonChilren.type).toBe('span');
    expect(buttonChilren.props.children).toEqual('Test Button');
  });

  it('has the passed id', () => {
    expect(button.getElement().props.id).toEqual('test');
  });

  it('responds to click event', () => {
    button.simulate('click');
    expect(mockClick.mock.calls.length).toEqual(1);
  });
});
