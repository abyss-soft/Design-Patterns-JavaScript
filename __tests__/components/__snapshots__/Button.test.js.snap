// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component renders with a DARK theme 1`] = `
.c0 {
  background-color: #484848;
  border: 1px solid #484848;
  border-radius: 4px;
  cursor: pointer;
  font: 400 1rem 'Karla','sans-serif';
  height: 48px;
  max-width: 240px;
  outline: none;
  padding: 0 1.5rem;
  width: 37.5%;
}

.c0 span {
  color: #F5F5F5;
}

.c0:hover {
  background-color: #888888;
  border-color: #888888;
}

.c0:hover span {
  color: #FFFFFF;
}

<button
  className="c0"
  id="test"
  onClick={[MockFunction]}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`<Button /> component renders with a LIGHT theme 1`] = `
.c0 {
  background-color: #F2E8F2;
  border: 1px solid #EDB8ED;
  border-radius: 4px;
  cursor: pointer;
  font: 400 1rem 'Karla','sans-serif';
  height: 48px;
  max-width: 240px;
  outline: none;
  padding: 0 1.5rem;
  width: 37.5%;
}

.c0 span {
  color: #6F256F;
}

.c0:hover {
  background-color: #6F256F;
  border-color: #6F256F;
}

.c0:hover span {
  color: #FFFFFF;
}

<button
  className="c0"
  id="test"
  onClick={[MockFunction]}
>
  <span>
    Test Button
  </span>
</button>
`;
